crates/api_common/src/site.rs

```diff
@@ -260,6 +261,8 @@ pub struct EditSite {
   pub allowed_instances: Option<Vec<String>>,
   /// A list of blocked instances.
   pub blocked_instances: Option<Vec<String>>,
+  /// A set of reasons, keyed by blocked instance ids
+  pub blocked_instances_reasons: Option<HashMap<String, String>>,
```

Comment:
This definitely needs to be reworked... Rather than having a separate map just for reasons. `blocked_instances: Option<Vec<BlockAction..`, where BlockAction contains the optional reason along with the instance domain string.

It will probably be a while on this then, as I'd want to get @SleeplessOne1917 's input.

Reply:
definitely agree in principle , this was me trying to not make a breaking API change as i'm assuming there's a whole versioning process for breaking changes?

Reply:
Yep this would be a breaking API change, but considering the lemmy API hasn't really been finalized or hit a `1.0` status yet, don't be afraid to do those.

Reply:
I don't think I'm familiar with the backend codebase or DB tables to give much useful insight on this TBH.

Reply:
See below.